R/getNeatOrder.R

```diff
@@ -0,0 +1,164 @@
+#' @param centering A single \code{character} value specifying the method to center the data. Options are \code{mean}, \code{median}, or \code{none} if your data is already centered. (default: method = \code{"mean"})
+#' @param ... Additional arguments passed to other methods.
+#' @return A \code{character} vector of row names in the sorted order.
+#' 
+#' @details 
+#' It's important to note that the [\pkg{sechm}](https://bioconductor.org/packages/3.18/bioc/vignettes/sechm/inst/doc/sechm.html#row-ordering) package does actually have the functionality 
+#' for plotting a heatmap using this radial theta angle ordering, though only by using an 
+#' MDS ordination. 
+#'   
+#' That being said, the \code{getNeatOrder} function is more modular and separate to the plotting, and 
+#' can be applied to any kind of ordinated data which can be valuable depending on the use case.
+#' 
+#' [Rajaram & Oono (2010) NeatMap - non-clustering heat map alternatives in R](https://doi.org/10.1186/1471-2105-11-45) outlines this in more detail.
+#' 
+#' @name getNeatOrder
+#' 
+#' @examples
+#' ## Load the required libraries and dataset
+#' library(mia)
+#' library(scater)
+#' library(ComplexHeatmap)
+#' library(circlize)
+#' data(peerj13075)
+#' 
+#' ## Group data by taxonomic order
+#' tse <- agglomerateByRank(peerj13075, rank = "order", onRankOnly = TRUE)
+#' 
+#' ## Transform the samples into relative abundances using CLR
+#' tse <- transformAssay(tse, assay.type = "counts", method="clr", MARGIN = "samples", name="clr", pseudocount = TRUE)
+#' 
+#' ## Transform the features (taxa) into zero mean, unit variance (standardize transformation)
+#' tse <- transformAssay(tse, assay.type="clr", method="standardize", MARGIN = "features", name="standardized")
+#' 
+#' ## Perform PCA using calculatePCA
+#' res <- calculatePCA(tse, assay.type = "standardized", ncomponents = 10)
+#' 
+#' ## Sort by radial theta and sort the original assay data
+#' sorted_order <- getNeatOrder(res[, c(1,2)], centering = "mean")
+#' tse <- tse[, sorted_order]
+#' 
+#' ## Define the color function and cap the colors at [-5, 5]
+#' col_fun <- colorRamp2(c(-5, 0, 5), c("blue", "white", "red"))
+#' 
+#' ## Create the heatmap
+#' heatmap <- Heatmap(assay(tse, "standardized"),
+#'               name = "NeatMap",
+#'               col = col_fun,
+#'               cluster_rows = FALSE,  # Do not cluster rows
+#'               cluster_columns = FALSE,  # Do not cluster columns
+#'               show_row_dend = FALSE,
+#'               show_column_dend = FALSE,
+#'               row_names_gp = gpar(fontsize = 4), 
+#'               column_names_gp = gpar(fontsize = 6), 
+#'               heatmap_width = unit(20, "cm"),  
+#'               heatmap_height = unit(15, "cm")  
```

Comment:
Is there a reason to not use sechm package to simplify the plotting?

Reply:
I think Leo had asked in a previous comment to use ComplexHeatmap, with this in mind should I still change the OMA example plot to sechm?

Reply:
ComplexHeatmap is fine here. I think in OMA we should use sechm whenever it is possible to streamline the analysis (for example plotting abundances). However, sechm is not suitable for every situation which is why we have to have some examples on ComplexHeatmap also (for example when plotting correlation heatmap). The nice thing is that sechm is just a wrapper for ComplexHeatmap so both approaches give the result in same format

Reply:
ok thanks sounds good